### PR TITLE
Release 2.7.0

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
# rc1

## Debs

Test plan: https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir

### Installed from from ppa:mir-team/rc

Platform|22.04LTS|21.10|20.04LTS
--|--|--|--
gbm-kms|n/a||
gbm-x11|n/a||
eglstream-kms|n/a||
dispmanx|n/a||

Console|22.04LTS|21.10|20.04LTS
--|--|--|--
minimal|n/a||
linux-VT|n/a||
logind|n/a||

WM|22.04LTS|21.10|20.04LTS
--|--|--|--
floating|n/a||
tiling|n/a||
kiosk|n/a||

Note, we don't have 22.04 builds in [the PPA](https://launchpad.net/~mir-team/+archive/ubuntu/rc) because of [lp:1958984](https://bugs.launchpad.net/ubuntu/+source/gcc-defaults/+bug/1958984)

**Comment by wmww** [comment]
I will upgrade to 21.10 tonight and test on that OS tomorrow.

**Comment by RAOF** [comment]
Hm. I don't have easy access to a pre-22.04 system. I'll see if there's anything I can do to trace down the FTBFS.

**Comment by Saviq** [comment]
All the lab tests for 2.7.0-rc1 succeeded 👍 .

**Comment by Saviq** [comment]
> Note, we don't have 22.04 builds in [the PPA](https://launchpad.net/~mir-team/+archive/ubuntu/rc) because of [lp:1958984](https://bugs.launchpad.net/ubuntu/+source/gcc-defaults/+bug/1958984)

Shall we cherry-pick #2320?

**Comment by wmww** [comment]
Note #2324. Introduced in last release, not this one, but potentially worth fixing.

**Comment by wmww** [comment]
#2247 appears to be back.

**Comment by wmww** [comment]
#2247 is fine now.  It seems our first rc wasn't sending `.axis_stop`? for some reason??? But after an upgrade it's sending the event and FF works as expected. Maybe a libinput compat problem, who knows.

**Comment by AlanGriffiths** [comment]
> #2247 is fine now. It seems our first rc wasn't sending `.axis_stop`? for some reason??? But after an upgrade it's sending the event and FF works as expected. Maybe a libinput compat problem, who knows

There's no way the rc1-rc2 changes are relevant. See the diff - only jammy has any changes. There must be another, unnoticed, factor.

**Comment by RAOF** [comment]
I ran into #2325 while testing; I don't know if this is a regression or not yet.

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,7 +29,7 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 set(MIR_VERSION_MAJOR 2)
 set(MIR_VERSION_MINOR 6)
-set(MIR_VERSION_PATCH 0)
+set(MIR_VERSION_PATCH 1)
```
Comment (RAOF) [inline CMakeLists.txt:32]: I think we've decided this should be 2.7.0, not 2.6.1, right? It's got feature additions.

Regardless, this should match what's in debian/changelog

**Review by RAOF — DISMISSED** [review]

**Comment by RAOF** [comment]
I've additionally tested with egmde-155-mir2.7.0~rc1-snap188, and cannot reproduce #2247.

**Comment by RAOF** [comment]
Another egmde-and-CLion bug: #2326.

I'll check if these bugs are regressions tomorrow morning.

**Comment by AlanGriffiths** [comment]
# rc3

## Debs

Test plan: https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir

### Installed from from ppa:mir-team/rc

Platform|22.04LTS|21.10|20.04LTS
--|--|--|--
gbm-kms|||@AlanGriffiths
gbm-x11|||@AlanGriffiths
eglstream-kms|||
dispmanx|||

Console|22.04LTS|21.10|20.04LTS
--|--|--|--
minimal|||@AlanGriffiths
linux-VT|||
logind|||@AlanGriffiths

WM|22.04LTS|21.10|20.04LTS
--|--|--|--
floating|||@AlanGriffiths
tiling|||@AlanGriffiths
kiosk|||@AlanGriffiths

**Comment by wmww** [comment]
PR to fix keyboard focus issues: #2330. Not sure if this is something we want to include here or not. Regression was introduced in last release.

**Comment by AlanGriffiths** [comment]
> PR to fix keyboard focus issues: #2330. Not sure if this is something we want to include here or not. Regression was introduced in last release.

We can consider cherry-picking once it lands on `main`. Given the history you describe in #2324, there's a significant risk of introducing additional issues. Some time for manual testing would mitigate that, as would improvements to automated testing. (I have seen you're already working on that - thanks!)

**Comment by AlanGriffiths** [comment]
# rc6

## Debs

Test plan: https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir

### Installed from from ppa:mir-team/rc

Platform|22.04LTS|21.10|20.04LTS
--|--|--|--
gbm-kms|@AlanGriffiths||
gbm-x11|@AlanGriffiths||
eglstream-kms|||
dispmanx|||

Console|22.04LTS|21.10|20.04LTS
--|--|--|--
minimal|@AlanGriffiths||
linux-VT|||
logind|@AlanGriffiths||

WM|22.04LTS|21.10|20.04LTS
--|--|--|--
floating|@AlanGriffiths||
tiling|||
kiosk|@AlanGriffiths||

**Comment by AlanGriffiths** [comment]
@RAOF rc6 has all the cherry-picks we've discussed. Please have a final look, I aim to tag tomorrow.

**Comment by RAOF** [comment]
Looks like #2330 has resulted in a regression - #2334. This affects both main an the 2.7 branch, and is a regression from 2.6

**Comment by AlanGriffiths** [comment]
I see another regression #2336

**Comment by AlanGriffiths** [comment]
OK, there's more to be investigated, but we can probably release.

OS|#2334|#2336
--|--|--
20.04|n/a|not with egmde/frame snaps
21.10|"fixed" by disabling zwp_text_input_v2|n/a
22.04|"fixed" by disabling zwp_text_input_v2|n/a

So,  #2336 seems potentially to be an issue with my test environment on 20.04.

Core20 based snaps are likely unaffected by #2334 as I don't see it on 20.04, but it may be prudent to disable by default and explicitly enable in Frame. Maybe add #2337?

**Comment by AlanGriffiths** [comment]
There's also  #2338 as a possible blocker

**Comment by RAOF** [comment]
Ok. #2334 occurs under kwin as well, so I'm pretty sure this is just qterminal being buggy, and there's nothing we need to do.

#2340 fixes #2338, and I think we should cherry-pick that.

**Comment by AlanGriffiths** [comment]
> #2340 fixes #2338, and I think we should cherry-pick that.

Sounds right, but I'm "not in" today and not at a computer. @Saviq?

**Comment by Saviq** [comment]
> Sounds right, but I'm "not in" today and not at a computer. @Saviq?

Yup. Will get it done.

**Comment by Saviq** [comment]
# rc7

## Debs

Test plan: https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir

### Installed from from ppa:mir-team/rc

Platform|22.04LTS|21.10|20.04LTS
--|--|--|--
gbm-kms|||
gbm-x11|||
eglstream-kms|||
dispmanx|||

Console|22.04LTS|21.10|20.04LTS
--|--|--|--
minimal|||
linux-VT|||
logind|||

WM|22.04LTS|21.10|20.04LTS
--|--|--|--
floating|||
tiling|||
kiosk|||

**Comment by wmww** [comment]
Uh oh, testing has reveled *yet another* focus bug. This time a rare race condition when opening weston-terminal. Pretty sure it's #2331, which was previously theoretical.

**Comment by wmww** [comment]
Another minor issue that doesn't seem to be a regression on this release, but was probably introduced somewhat recently: #2341, fixed by #2342.

**Comment by RAOF** [comment]
# rc9

## Debs

Test plan: https://wiki.ubuntu.com/Process/Merges/TestPlans/Mir

### Installed from from ppa:mir-team/rc

Platform|22.04LTS|21.10|20.04LTS
--|--|--|--
gbm-kms|||
gbm-x11|||
eglstream-kms|||
dispmanx|||

Console|22.04LTS|21.10|20.04LTS
--|--|--|--
minimal|||
linux-VT|||
logind|||

WM|22.04LTS|21.10|20.04LTS
--|--|--|--
floating|||
tiling|||
kiosk|||

**Comment by AlanGriffiths** [comment]
Only thing I've found with rc9 so far is that CLion can end up with weird focus states, but that's not blocking. (CLion has some weird focus behaviour on Mutter too, but not as bad.)

Will document when I have reproducers.

**Comment by AlanGriffiths** [comment]
bors merge

**Pull request merged**